core/src/rewards_service.rs

```diff
@@ -0,0 +1,68 @@
+use crossbeam_channel::{Receiver, RecvTimeoutError, Sender};
```

Comment:
Shoehorning this into the TransactionStatusService didn't feel right, and I wasn't sure what to call a unified Service so I just went for this

Reply:
Separate service wfm. Not sure about the name, though. A RewardsService sounds like the thing that actually bestows rewards, when actually it's just storing data about those rewards.
Is RewardsDataService too much?

Reply:
I'm trying `RewardsRecorderService` on for size

Reply:
I like that better